Guardian/Guardian.swift

```diff
@@ -187,6 +187,64 @@ public func notification(from userInfo: [AnyHashable: Any]) -> Notification? {
     return AuthenticationNotification(userInfo: userInfo)
 }
 
+/**
+ Registers the application to receive Guardian push notifications
+ 
+ Should be called in your AppDelegate's `application(:didFinishLaunchingWithOptions)` method:
+ 
+ ```
+ func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplicationLaunchOptionsKey: Any]?) -> Bool {
+    /* ... */
+
+    // Set up push notifications
+    Guardian.registerForRemoteNotification(application)
+ }
+ ```
+ 
+ Remember that you should also override your `application(:handleActionWithIdentifier:forRemoteNotification:withResponseInfo:completionHandler)` 
+ method to automatically handle the notification actions:
+ 
+ ```
+ func application(_ application: UIApplication, handleActionWithIdentifier identifier: String?, forRemoteNotification userInfo: [AnyHashable : Any], withResponseInfo responseInfo: [AnyHashable : Any], completionHandler: @escaping () -> Void) {
+    if let notification = Guardian.notification(from: userInfo) {
+        /* Get the enrollment that matches the notification ... */
+        let enrollment: Enrollment = ...
+        Guardian
+            .authentication(forDomain: "tenant.guardian.auth0.com", andEnrollment: enrollment)
+            .handleAction(withIdentifier: identifier, andNotification: notification)
+            .start { result in
+                completionHandler()
+        }
+    }
+ }
+ ```
+ */
+public func registerForRemoteNotifications(_ application: UIApplication) {
```

Comment:
Let's spin this in a different way.

Let's have a method to request the category, since it's more likely that the developer would want to customize the other notification settings so let's have a method like:

```swift
public func categoryForNotification(withAcceptTitle acceptTitle: String, rejectTitle: String) -> UIUserNotificationCategory
```